rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,299 @@
+            ) from e
+
+        token = get_ethereum_token(
+            symbol=token_symbol,
+            ethereum_address=token_address,
+            name=token_name,
+            decimals=token_decimals,
+        )
+        if token_total_amount == ZERO:
+            raise DeserializationError(f'Token {token.identifier} balance is zero')
+
+        token_user_amount = user_amount / total_amount * token_total_amount
+        weight = token_weight * 100 / total_weight
+        pool_token = BalancerPoolToken(
+            token=token,
+            total_amount=token_total_amount,
+            user_balance=Balance(amount=token_user_amount),
+            weight=weight,
+        )
+        pool_tokens.append(pool_token)
+
+    pool = BalancerPool(
+        address=address,
+        tokens=pool_tokens,
+        total_amount=total_amount,
+        user_balance=Balance(amount=user_amount),
+    )
+    return user_address, pool
+
+
+def deserialize_transaction_id(raw_tx_id: str) -> Tuple[str, int]:
+    try:
+        tx_hash, raw_log_index = raw_tx_id.split('-')
+        log_index = int(raw_log_index)
+    except ValueError as e:
+        raise DeserializationError(f'Unexpected transaction id: {raw_tx_id}') from e
+    return tx_hash, log_index
+
+
+def deserialize_swap(raw_swap: Dict[str, Any]) -> AMMSwap:
+    """May raise DeserializationError"""
+    try:
+        tx_hash, log_index = deserialize_transaction_id(raw_swap['id'])
+        timestamp = deserialize_timestamp(raw_swap['timestamp'])
+        raw_tokens = raw_swap['poolAddress']['tokens']
+        token0_symbol = raw_swap['tokenInSym']
+        token1_symbol = raw_swap['tokenOutSym']
+        amount0_in = deserialize_asset_amount(raw_swap['tokenAmountIn'])
+        amount1_out = deserialize_asset_amount(raw_swap['tokenAmountOut'])
+        raw_token_in = raw_swap['tokenIn']
+        raw_token_out = raw_swap['tokenOut']
+        raw_addresses = SwapRawAddresses(
+            user_address=raw_swap['userAddress']['id'],  # address
+            caller=raw_swap['caller'],  # from_address
+            pool_address=raw_swap['poolAddress']['id'],  # to_address
+            token_in=raw_token_in,  # token0_address
+            token_out=raw_token_out,  # token1_address
+        )
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    # Checksum addresses
+    try:
+        addresses = SwapAddresses(
+            user_address=to_checksum_address(raw_addresses.user_address),
+            caller=to_checksum_address(raw_addresses.caller),
+            pool_address=to_checksum_address(raw_addresses.pool_address),
+            token_in=to_checksum_address(raw_addresses.token_in),
+            token_out=to_checksum_address(raw_addresses.token_out),
+        )
+    except ValueError as e:
+        error_msg = str(e)
+        for field in raw_addresses._fields:
```

Comment:
I was trying to provide a detailed information about the string (raw address) and the failing swap key (its name) in case of failing to checksum, whilst avoiding 5 blocks of try...except. This is why first the dicts and then the named tuple. I've finally decided to checksum one by one, it doesn't add that many lines.